src/account/operations/transaction/high_level/create_alias.rs

```diff
@@ -0,0 +1,169 @@
+pub struct AliasOutputOptionsDto {
+    /// Bech32 encoded address which will control the alias. Default will use the first
+    /// address of the account
+    pub address: Option<String>,
+    /// Immutable alias metadata, hex encoded bytes
+    #[serde(rename = "immutableMetadata")]
+    pub immutable_metadata: Option<String>,
+    /// Alias metadata, hex encoded bytes
+    pub metadata: Option<String>,
+}
+
+impl TryFrom<&AliasOutputOptionsDto> for AliasOutputOptions {
+    type Error = crate::Error;
+
+    fn try_from(value: &AliasOutputOptionsDto) -> crate::Result<Self> {
+        Ok(Self {
+            address: value.address.clone(),
+            immutable_metadata: match &value.immutable_metadata {
+                Some(metadata) => {
+                    Some(prefix_hex::decode(metadata).map_err(|_| DtoError::InvalidField("immutable_metadata"))?)
+                }
+                None => None,
+            },
+            metadata: match &value.metadata {
+                Some(metadata) => Some(prefix_hex::decode(metadata).map_err(|_| DtoError::InvalidField("metadata"))?),
+                None => None,
+            },
+        })
+    }
+}
+
+impl AccountHandle {
+    /// Function to create an alias output.
+    /// ```ignore
+    /// let alias_options = AliasOutputOptions {
+    ///     address: None,
+    ///     immutable_metadata: Some(b"some immutable alias metadata".to_vec()),
+    ///     metadata: Some(b"some alias metadata".to_vec()),
+    /// };
+    ///
+    /// let transaction = account.create_alias_output(alias_options, None).await?;
+    /// println!(
+    ///     "Transaction: {} Block sent: http://localhost:14265/api/core/v2/blocks/{}",
+    ///     transaction.transaction_id,
+    ///     transaction.block_id.expect("no block created yet")
+    /// );
+    /// ```
+    pub async fn create_alias_output(
+        &self,
+        alias_output_options: Option<AliasOutputOptions>,
+        options: Option<TransactionOptions>,
+    ) -> crate::Result<Transaction> {
+        let prepared_transaction = self.prepare_create_alias_output(alias_output_options, options).await?;
+        self.sign_and_submit_transaction(prepared_transaction).await
+    }
+
+    pub(crate) async fn prepare_create_alias_output(
+        &self,
+        alias_output_options: Option<AliasOutputOptions>,
+        options: Option<TransactionOptions>,
+    ) -> crate::Result<PreparedTransactionData> {
+        log::debug!("[TRANSACTION] prepare_create_alias_output");
+        let rent_structure = self.client.get_rent_structure().await?;
+
+        let controller_address = match alias_output_options
+            .as_ref()
+            .and_then(|options| options.address.as_ref())
+        {
+            Some(bech32_address) => Address::try_from_bech32(bech32_address)?.1,
+            None => {
+                self.public_addresses()
+                    .await
+                    .first()
+                    .ok_or(Error::FailedToGetRemainder)?
```

Comment:
Just checking, correct error ?

Reply:
Not really, in other places we just unwrap since the first address is generated during account creation, should I do that here also or add a new error for this case?

Reply:
If there is absolutely no way to create an account with 0 address then yeah I'd unwrap + comment or expect